plugins/module_utils/ansible_freeipa_module.py

```diff
@@ -1169,6 +1169,28 @@ def ipa_command_param_exists(command, name):
         """
         return api_check_param(command, name)
 
+    def ipa_command_param_has_choice(self, command, name, value):
```

Comment:
As most calls of this method are done in a loop to test all values of a paramter, what do you think about having a method that acts like:
```python
def ipa_command_invalid_param_choices(self, command, name, value):
    return (set(value) - set([""])) - set(api.Command[command].params[name].cli_metavar)
```

With a method like this, we could pass a list of the options and if the return is not empty, we would know which values are not supported.

The client code could be easier to follow with such a method:
```python
invalid_options = module.ipa_command_invalid_param_choices(cmd, param, [...])
if "x" not in invalid_options:
   raise Exception()
if "y" not in invalid_options:
   raise Exception()
```

Reply:
Fixed